projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitActivity.java

```diff
@@ -184,121 +186,64 @@ private String fillTemplate() throws IOException {
-        lawsuit.replace(
-            "<companyFax>", ((EditText) findViewById(R.id.companyFax)).getText().toString());
-
+    lawsuit = lawsuit.replace("<undefined>", "");
     return lawsuit;
   }
 
-  // Pops a dialog asking the user to share the pdf
-  private void sharePdf(String absFilename) {
-    if (new File(absFilename).exists()) {
-      new AlertDialog.Builder(this, R.style.AlertDialog)
-          .setTitle("כתב תביעה נוצר")
-          .setMessage("כתב התביעה נוצר ונשמר במכשירך.\n האם תרצה/י לשתף?")
-          .setPositiveButton(
-              "כן",
-              (dialog, which) -> {
-                Uri uri =
-                    FileProvider.getUriForFile(
-                        getApplicationContext(),
-                        getPackageName() + ".fileprovider",
-                        new File(absFilename));
-                Intent shareIntent = new Intent();
-                shareIntent.setAction(Intent.ACTION_SEND);
-                shareIntent.setType("application/pdf");
-                shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
-                startActivity(Intent.createChooser(shareIntent, "Share via"));
-              })
-          .setNegativeButton("לא", (dialog, which) -> dialog.dismiss())
-          .create()
-          .show();
+  // Create a folder for generated PDFs (external storage)
+  private void createOutputDir() {
+    File directory =
+        new File(
+            Paths.get(
+                    Environment.getExternalStorageDirectory().getPath(),
+                    getString(R.string.app_name),
+                    getString(R.string.output_folder_name))
+                .toString());
+    if (!directory.exists()) {
+      directory.mkdirs();
     }
-    Log.w(TAG, "Could not share pdf, file doesn't exists.\nabsFilename: " + absFilename);
-  }
-
-  // Pops a calendar dialog when clicking on date fields
-  private void displayDatePicker() {
-    calendar = Calendar.getInstance();
-    int day = calendar.get(Calendar.DAY_OF_MONTH);
-    int month = calendar.get(Calendar.MONTH);
-    int year = calendar.get(Calendar.YEAR);
-
-    datePickerDialog =
-        new /**/ DatePickerDialog(
-            LawsuitActivity.this,
-            (datePicker, currentYear, currentMonth, currentDay) ->
-                receivedDate.setText(day + "/" + (month + 1) + "/" + year),
-            year,
-            month,
-            day);
-    datePickerDialog.show();
   }
 
+  // Flags 'permissionGranted" for results
   private void checkPermissions() {
     // Ask runtime permissions for devices running SDK > 22
     if (Build.VERSION.SDK_INT >= 23) {
 
       // Permission granted
-      if ((checkSelfPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
+      if ((ActivityCompat.checkSelfPermission(
+              LawsuitActivity.this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
           == PackageManager.PERMISSION_GRANTED)) {
         permissionGranted = true;
         createOutputDir();
```

Comment:
This function check for permissions, it is not suppose to be the one to create the directory in storage, create it when it needed, before you try to write into it.

Reply:
Added a TODO for this one. I'll have it fixed next round.